boot/freeldr/freeldr/arch/i386/hwacpi.c

```diff
@@ -87,8 +94,7 @@ DetectAcpiBios(PCONFIGURATION_COMPONENT_DATA SystemKey, ULONG *BusNumber)
         AcpiPresent = TRUE;
 
         /* Calculate the table size */
-        TableSize = PcBiosMapCount * sizeof(BIOS_MEMORY_MAP) +
-            sizeof(ACPI_BIOS_DATA) - sizeof(BIOS_MEMORY_MAP);
+        TableSize = FindAcpiTableSize();
 
```

Comment:
let me know if I'm correct, but it seems that the FindAcpiBios() call made above, plus the Rsdp usages made below there (lines 116/122), look awfully like what FindAcpiTable() does.

It seems that FindAcpiTable() function could be modified to becomes this:
```c
PVOID
FindAcpiTable(
    _Out_opt_ PSIZE_T TableSize);
```
which would return in `*TableSize` what your proposed `FindAcpiTableSize()` returns.

Reply:
i mean i guess, sure why not

Reply:
Id rather not do this, it requires a bunch of changes to existing code